### ISSUE TITLE
Firewalls, shortcuts and updates

### ISSUE THREAD

**Issue by Cryo99** [duplicati/duplicati#4154]
- [X] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.5.104_canary_2020-03-25
- **Operating system**: Win10
- **Backend**: Google Drive

## Description

Yesterday I was investigating why, yet again, Duplicati had stopped making backups. I have a shortcut (set to run as admin) in \<start menu\>\start-up which doesn't work. I'd end up starting `Duplicati.GUI.TrayIcon.exe` manually as an admin and it would run until the next restart, when it would fail to start again. In addition, even when running, it would sometimes fail to connect until I poked around for a while and re-started things and then it would, usually, get going again. One common theme was that Windows firewall would regularly decide to block the connection, which would cause the failure-to-backup problem.

So, yesterday I updated in response to the normal prompt and lo and behold, no backups. Poke, poke, restart, nope, bloody firewall again, let it through, back to sanity. Having had enough of this I looked more closely at what was running and why and discovered two instances of `Duplicati.GUI.TrayIcon.exe` running. That's odd. Check paths. Oooooooooooh. Kill the one that's running from where it _should_ be running. It still works. What the hell's going on here. I poke around some more and discover the `updates` directory....

So now I see the light. Duplicati has 'installed' several versions, i.e. the updates, into a sub-directory of the location I put it in. It's now running from one of these locations and breaking the firewall rules.  
I killed the copy running from the original location and tried running backups. No problem, so that's redundant. Seeking an explanation for all of this this, I found the following relevant entry in the blog:

> Which one? As mentioned, nothing in C:\Program Files\Duplicati 2 is supposed to be updated.
> Instead, C:\ProgramData\Duplicati\updates\2.0.5.1 is supposed to get an entirely new version.
>
>Starting Duplicati starts the Program Files version, which then locates and starts the new one.

So that's the problem. Each update folder has a complete copy (in my case there are three!)  of the application and all of it's dependencies. Copies are being run, without my knowledge or permission, from locations that they weren't told to run from. Redundant instances are being kept in memory. None of this explains why it stops running from the shortcut but it raises many questions.

1. What on earth is going on here?
2. Why are you using 4 times the amount of disk space necessary to store the application?
3. Why are you running multiple instances of `Duplicati.GUI.TrayIcon.exe` unnecessarily?
4. Why are you running the application from locations other than the one I chose, breaking system configuration?

## Steps to reproduce

1. Run it and update.

- **Actual result**: 
Madness.
- **Expected result**:
1. Duplicati should not use more space than necessary to store one copy of itself.
2. Duplicati should update that one copy of itself during updates, like every other application ever written in the software development history of mankind. Probably.
3. Duplicati should not run from anywhere except where I tell it to run from.
4. Duplicati should never, ever break configuration I create to accommodate it.
5. Ever!

## Screenshots

N/A

## Debug log

N/A

**Comment by drwtsn32x** [comment]
Do you have your Windows firewall configured to block application outbound communication by default?

I believe the reason the auto update mechanism works like this is so non-privileged users can apply updates.  Personally I'm not a huge fan of it and don't use it - using it is not required.  You can just install updates manually by downloading the MSI package.  When you use this approach, the main program files are updated as you'd expect.

**Comment by warwickmm** [comment]
As @drwtsn32x mentions, this is by design so that non-privileged users can apply updates.  This auto-update can be [disabled](https://duplicati.readthedocs.io/en/latest/07-other-command-line-utilities/#duplicatilibraryautoupdaterexe), and you can update manually instead.

**Comment by Cryo99** [comment]
I switched to the installer but it was still unreliable and stopped running several times so I had to dump it. I appreciate the work that's gone into it and really did think that I'd finally found the backup software I'd been craving for all these years with Duplicati but the upgrade design is flawed, it's unreliable and I spent far too much time babying it, trying to find out why it had stopped running yet again, having to fix the DB, etc. and I don't have the time or energy for that. Maybe when things have improved but for now it's back to the drawing board. 😞 😢
